### PR TITLE
[Week4]-jaeyeon

### PR DESCRIPTION
**Pull request by seojae02**
심층 신경망에선 다음과 같은 문제가 발생할 확률이 높다. 
1. 그래디언트 소실(vanishing gradient) 또는 폭주(exploding)가 발생할 수 있다.
2. 모델이 복잡하고 커질수록 학습시간이 매우 느려진다.
3. 모델이 복잡할수록 오버피팅(overfitting)될 위험이 크다.

신경망 학습에서 중요한 것은 학습 시킬 떄의 가중치 초기값이다. 가중치 초기값을 어떻게 초기화 하느냐에 따라 학습의 성능이 달라진다.
가중치의 초기값을 모두 0으로 초기화하거나 동일한 값으로 초기화할 경우 모든 뉴런의 동일한 출력값을 내보낸다. 가중치 초기값은 작은 값으로 초기화 해야하는데 그 이유는 활성화 함수가 sigmoid일 경우 만약 가중치 초기값(절대값)을 큰 값으로 한다면 0과 1로 수렴하기 때문에 그래디언트 소실이 발생하게 된다.

인공신경망
퍼셉트론 :1957년 프랭크 로젠블라트가 제안한 가장 기본적인 인공 신경망 모델
선형 문제만 해결 가능. XOR같은 비선형 문제를 해결할 수 없다는 한계 존재 >> 다층 퍼셉트론 (MLP) , 역전파 알고리즘 (Backpropagation)

### CODE REVIEW

**Review by kimhamyong — APPROVED** [review]
task1: 잘 수행하셨습니다.

**Review by giyomdabin — APPROVED** [review]
task2: 주어진 과제를 잘 수행하셨습니다. 수고하셨습니다!

**Comment by kimhamyong** [comment]
task1: 

추가적으로, 학습 도중 검증 손실이 더 이상 감소하지 않을 때 학습을 조기 종료하도록 EarlyStopping 콜백을 추가하면 불필요한 에포크 실행을 방지할 수 있습니다. (필수 사항 X)

   ```python
from tensorflow.keras.callbacks import EarlyStopping
early_stopping = EarlyStopping(monitor='val_loss', patience=3, restore_best_weights=True)
history = model.fit(x_train, y_train, batch_size=64, epochs=50, validation_split=0.2, callbacks=[early_stopping])
   ```
